### DISCUSSION TITLE
Why does Alpine3 creates such a huge degradation in performance ?

### DISCUSSION THREAD

**Discussion by magneto0007** [alpinejs/alpine#4462] [1. Help]
I have noticed countless Hyva theme site having very bad performance score using https://pagespeed.web.dev/.

They all point to a huge execution time by alpine3.

Thanks

**Comment by SimoTod** [comment]
You should ask who wrote the theme.

In general, Alpine runs after the page loads so it's obviously slower than bare html.
On top of that, it needs to crawl all tags inside a component to find its directives. Note, it won't crawl the full page, just what's inside a tag with the x-data attribute but many people lazily put an x-data on the body tag.

That being said, if the slowness is not noticeable by a human being, pagespeed is just a score.

**Comment by ekwoka** [comment]
Would need to see what the site is.

The performance is competitive, but it is slower than more raw JS solutions, because it has some overhead just to handle the dom as the source of truth.

but any specific site being slow doesn't seem like alpine should be that relevant.

I have some pretty wild Alpine sites that don't have alpine being a cause of issue.
